src/components/apex-chart/apex-chart.ts

```diff
@@ -0,0 +1,81 @@
+import {bindable} from "aurelia-templating";
+import ApexCharts, {ApexOptions} from 'apexcharts';
+
+export interface ISelection {
+    series?:number;
+    dataPointIndex?:number;
+}
+
+export type ApexSeries = ApexAxisChartSeries | ApexNonAxisChartSeries
+
+export class ApexChart {
+    private _apex: HTMLDivElement;
+    private _myApexChart: ApexCharts;
+
+    @bindable options: ApexOptions;
+    @bindable series: ApexSeries;
+    @bindable selection: ISelection;
+    @bindable class:string;
+
+    private _attached:boolean = false;
+
+    attached() {
+        this._attached = true;
+        if (this.options) {
+            this._createChart();
+        }
+
+    }
+
+    detached() {
+        this._attached = false;
+        if (this._myApexChart) {
+            this._myApexChart.destroy();
+            this._myApexChart = null;
+        }
+    }
+
+    optionsChanged(newOptions:ApexOptions) {
+        if (this._myApexChart) {
+            this._myApexChart.updateOptions(newOptions);
+        } else {
+            this._createChart();
+        }
+    }
+
+    seriesChanged(newSeries:ApexSeries) {
+        if (this._myApexChart) {
+            this._myApexChart.updateSeries(newSeries);
+        } else {
+            this._createChart();
+        }
+    }
+
+    selectionChanged(newValue:ISelection, oldValue:ISelection) {
+        if (this._myApexChart) {
+            if (newValue) {
+                this._myApexChart.toggleDataPointSelection(newValue.series, newValue.dataPointIndex);
+            }
+            else if (oldValue) {
+                this._myApexChart.toggleDataPointSelection(oldValue.series, oldValue.dataPointIndex);
+            }
+        }
+    }
+
+    private _createChart() {
+        if (!this._apex) {
+            return;
+        }
+
+        if (!this.options.series) {
```

Comment:
Might be a matter of preference, but I'd rather change the `options` binding to be one-way and simply update the series field there, instead of having to juggle two fields.
E.g. if the binding updates `this.options` and it contains a new series, should that not supercede `this.series`? Conversely, if the binding updates `this.series` should it not supercede `this.options.series`? Surely we want to always display the newer one - but we have no information about which one is newer!
(And more to the point: We _never_ want to display more than one, so we should not _store_ more than one.)

Reply:
The idea is to not having two fields. You update `options` first and `series` asynchronously later. But you can pass `series` within `options` in the first place and when passing `options` without new `series` it reuses the previous. There is actually nothing confusing.

Reply:
Mmh. Reads to me as if we'd actually want a field `data: ApexOptions | ApexSeries` instead?
Or possibly have no `series` field but a synthetic `series` setter that directly modifies `this.options`? (Dunno if Aurelia bindings support that, though.)

Give it a few months and I'd not feel confident to discern the expected inputs without checking the code. ... But that might be just me. It's not a dealbreaker.

Reply:
These bindings just cover the features that ApexCharts have:
```typescript
this.chart.updateOptions(newOptions);
this.chart.updateSeries(newSeries);
```
and that's the only thing I want. Plus the comfort to reuse existing series data when setting new options without explicit new series data.

Reply:
Fair enough.